dulwich/object_store.py

```diff
@@ -1265,7 +1266,7 @@ def commit_tree_changes(object_store, tree, changes):
     nested_changes = {}
     for (path, new_mode, new_sha) in changes:
         try:
-            (dirname, subpath) = path.split(b'/', 1)
```

Comment:
Paths passed in here should always be using / as a separator; it's the responsibility of the caller to convert them to git-style paths (with a /)